### PR TITLE
qt: replace BlockingWaitingDialog with RunCoroutineDialog

### PR DESCRIPTION
**Pull request by ecdsa**
RunCoroutineDialog has a run() method that blocks the thread without blocking the GUI (using exec), and a Cancel button that cancels the coroutine.

main_window.run_coroutine_dialog() is a wrapper that returns the coroutine result and may raise exceptions.

BlockingWaitingDialog was removed is transaction_dialog, where it was not particularly useful.

### CODE REVIEW

File: electrum/gui/qt/confirm_tx_dialog.py [hunk 1]
```diff
@@ -619,7 +619,7 @@ def __init__(self, *, window: 'ElectrumWindow', make_tx, output_value: Union[int
             title=_("New Transaction"), # todo: adapt title for channel funding tx, swaps
             allow_preview=allow_preview)
 
-        BlockingWaitingDialog(window, _("Preparing transaction..."), self.update)
```
Comment (SomberNight) [inline electrum/gui/qt/confirm_tx_dialog.py:622]: > BlockingWaitingDialog was removed is transaction_dialog, where it was not particularly useful.

I have a testnet wallet where `wallet.make_unsigned_transaction()` takes 17 seconds (which is called from self.update).
That was actually the motivation I added this wrapper in https://github.com/spesmilo/electrum/pull/5833 IIRC

Comment (ecdsa) [reply electrum/gui/qt/confirm_tx_dialog.py:622]: why is that useful? if it takes 17 seconds to sign the transaction, then BlockingWaitingDialog is going to block the GUI as well...

Comment (SomberNight) [reply electrum/gui/qt/confirm_tx_dialog.py:622]: The point of BlockingWaitingDialog was to give feedback to the user about what is going on. If you remove it without a replacement, the GUI still freezes, but now without even telling the user why.

Comment (ecdsa) [reply electrum/gui/qt/confirm_tx_dialog.py:622]: fixed in 6635e40a2c0d89ef68c7964d205f94287448f8aa

**Pull request merged**